mrrc/pkgs/npm.py

```diff
@@ -14,236 +14,61 @@
-    dist_tags: Optional[dict] = field(default_factory=dict)
-    versions: Optional[dict] = field(default_factory=dict)
-    maintainers: Optional[list] = field(default_factory=list)
-    description: Optional[str] = field(default="")
-    time: Optional[dict] = field(default_factory=dict)
-    author: Optional[str] = field(default="")
-    users: Optional[dict] = field(default_factory=dict)
-    repository: Optional[dict] = field(default_factory=dict)
-    readme: Optional[str] = field(default="")
-    readmeFilename: Optional[str] = field(default="")
-    homepage: Optional[str] = field(default="")
-    keywords: Optional[list] = field(default_factory=list)
-    bugs: Optional[str] = field(default="")
-    license: Optional[str] = field(default="")
-
-    def get_name(self):
-        return self.name
-
-    def set_description(self, description: str):
-        self.description = description
-        return self
-
-    def set_dist_tags(self, dist_tags: dict):
-        self.dist_tags = dist_tags
-        return self
-
-    def set_versions(self, versions: dict):
-        self.versions = versions
-        return self
-
-    def set_maintainers(self, maintainers: list):
-        self.maintainers = maintainers
-        return self
-
-    def set_time(self, time: dict):
-        self.time = time
-        return self
-
-    def set_author(self, author: str):
-        self.author = author
-        return self
-
-    def set_users(self, users: dict):
-        self.users = users
-        return self
-
-    def set_repository(self, repository: dict):
-        self.repository = repository
-        return self
-
-    def set_readme(self, readme: str):
-        self.readme = readme
-        return self
-
-    def set_readmeFilename(self, readmeFilename: str):
-        self.readmeFilename = readmeFilename
-        return self
-
-    def set_homepage(self, homepage: str):
-        self.homepage = homepage
-        return self
-
-    def set_keywords(self, keywords: list):
-        self.keywords = keywords
-        return self
-
-    def set_bugs(self, bugs: dict):
-        self.bugs = bugs
-        return self
-
-    def set_license(self, _license: str):
-        self.license = _license
-        return self
+    def __init__(self, version_metadata):
+        self.name = version_metadata.get('name')
+        self.description = version_metadata.get('description')
+        self.author = version_metadata.get('author')
+        self.license = version_metadata.get('license')
+        self.repository = version_metadata.get('repository')
+        self.bugs = version_metadata.get('bugs')
+        self.keywords = version_metadata.get('keywords')
+        self.maintainers = version_metadata.get('maintainers')
+        self.homepage = version_metadata.get('homepage')
+        self.dist_tags = {'latest': version_metadata.get('version')}
+        self.versions = {version_metadata.get('version'): version_metadata}
+        self.readme = version_metadata.get('readme', '')
```

Comment:
It's possible for the version metadata that doesn't have some meta values here, I'm not sure whether this could cover the empty or null case, and result in some type error when trying to do deserialization, finally we need to ensure the package version object is JSON serializable. @hjmodi @ligangty We need to confirm on this.

Reply:
@yma88 Is it possible to cover these cases in unittests?

Reply:
Should be, it's better to see whether there are some nil str popping up in json data, if @hjmodi could confirm no problem, I'm ok with it.

Reply:
@hjmodi @ligangty I just tested this PR locally, and unit tests could pass without non-standard JSON values. I pushed a commit (https://github.com/ligangty/mrrc-uploader/pull/1/commits/c50fbc70bbd26a144a873dd74648ae38cb09c1ef) based on your code which adds some missing fields and will make the attributes of non-existed value invisible in the package JSON file, this might be more in accord with the expectation.